### ISSUE TITLE
Convbin of RTCM3 file writes timestamps in the future

### ISSUE THREAD

**Issue by luc-girod** [tomojitakasu/RTKLIB#746]
Hello!

I have been using Convbin from 2.4.3 to convert files from my DJI D-RTK2 for a while now. It has been totally fine until today, when I tried to convert files from last week, and the obs times are written as if they were from this week (should be **2024    02    26 **, is **2024    03    04 **).

It's pretty likely a DRTK2 problem, or some kind of disagreement on week numbering, but I thought I'd mention it.
Attached is one of the .rtcm3 and converted .obs
[RTK_PPPtest.zip](https://github.com/tomojitakasu/RTKLIB/files/14500793/RTK_PPPtest.zip)

**Comment by yydgis** [comment]
it is rtcm file, right? use convbin -r rtcm3 -tr 2024/02/06 0:0:0 -v 3.04 rtcmfilename

**Comment by luc-girod** [comment]
Thanks @yydgis , that worked, but I have to run it with the right date (nearly) for each file, or I get 1 week errors :/

**Comment by yydgis** [comment]
yes, need to get the correct date, otherwise, the program will use the current date. if your rtcm file also log the ephemeris, it should be someway to recover the apprximate date/time, but the current convbin do not support it. The good practice is to have the time/date in the filename, then you will know the approximate date/time when you see the filename.

**Comment by gpoleszuk** [comment]
Have you tried this rtcm3 to rinex data conversion online service https://process.gps-solutions.com/index.html? It is a bit more advanced in terms of conversion, but it still requires your attention to the recording raw data timestamp. As the author mentioned, it is important that the file name contains the timestamp to facilitate the conversion without needing to infer this information from navigation files or processing.
